runtime/auth/aws-credentials/common/src/aws/smithy/kotlin/runtime/auth/awscredentials/CredentialsProvider.kt

```diff
@@ -4,12 +4,19 @@
  */
 package aws.smithy.kotlin.runtime.auth.awscredentials
 
+import aws.smithy.kotlin.runtime.io.Closeable
+
 /**
  * Represents a producer/source of AWS credentials
  */
-public interface CredentialsProvider {
+public interface CredentialsProvider : Closeable {
     /**
      * Request credentials from the provider
      */
     public suspend fun getCredentials(): Credentials
+
+    /**
+     * Shutdown and cleanup any resources
+     */
+    override fun close() { }
```

Comment:
**Question**: Why is this necessary? Won't this remove the need for implementations to implement `close`, potentially leading to resources that are never closed?

Reply:
see https://github.com/awslabs/smithy-kotlin/pull/776#discussion_r1072877830